sdk/include/psdk/d3dtypes.h

```diff
@@ -27,9 +27,12 @@
 
 /* #include <windows.h> FIXME: Need to include for compatibility. Inclusion caused compile fail */
 
-#if (DIRECT3D_VERSION >= 0x0800)
-#error "You should not include d3dtypes.h when compiling for DX8 or newer."
-#endif
+/* //FIXME: Need to define for compatibility. Defining causes compile fail with ReactOS d3d9 enabled.
+ *
+ * #if (DIRECT3D_VERSION >= 0x0800)
+ * #error "You should not include d3dtypes.h when compiling for DX8 or newer."
+ * #endif
+ */
```

Comment:
This seems like a huge hack?

Reply:
I guess yes. According to the defined error, d3dtypes.h should not be included in d3d9 (directly or from other headers), but for some reason it is included.

Reply:
Can you fix the include instead? I don't like hacking headers just to be able to build something that is broken.

Reply:
Sorry, I can't. I already tried that. If uncomment that definition and remove some headers which include d3dtypes.h (directly or from other headers) (e. g. d3dhal.h and d3dnthal.h) from d3d9 sources, compilation fails for me again with a lot of errors due to undeclared variables, incompatible argument/pointer types etc. It's because that headers have some definitions used by d3d9 sources. If I'll remove d3dtypes.h inclusion from all sdk headers which have it, I'll broke the headers structure and compilation will fail anyway. It seems to me that our d3dtypes.h, which is taken from Wine (with ROS-specific diffs), is a bit different that MS's one, so it needs more fixing. But in case if it is not a source of problem, only rewriting some wrong code parts of our d3d9 may be as an option. So this hack is the simplest compilation fix for now. I already said about this in the PR description.